src/main/rx/rx.c

```diff
@@ -338,6 +341,35 @@ void resumeRxSignal(void)
     failsafeOnRxResume();
 }
 
+#ifdef USE_OSD
+#define FRAME_QUALITY_SCALE 16
+#endif
+
+static void setFrameQuality(bool validFrame)
+{
+#ifdef USE_OSD
+    static uint8_t lastMissedCount;
+    static uint8_t currentMissedCount;
+
+    if (validFrame) {
```

Comment:
- I introduced FRAME_QUALITY_SCALE so that the max value could be fixed/unaffected by changes to the max missing frames we count up to.
- I'm not sure what the lowest common denominator across txs is for number of missing frames before treated as a failsafe but FRAME_QUALITY_SCALE should be set to something representative of that. I did wonder if I needed to average the calculated frame rate. That, then, would just give back the original averaging.

Reply:
Since `FRAME_QUALITY_SCALE` is just another constant all it does is make the code harder to read, for no benefit.
As I said, I think that the averaged percentage of dropped frames is a better indicator for link quality than the length of the last gap. For anything that creates periodic gaps (interference with a fixed frequency signal for example), this will just jump around and be unusable.